neon_homeassistant_skill/__init__.py

```diff
@@ -16,39 +17,112 @@ def chunks(lst, n_len) -> List[list]:
 class NeonHomeAssistantSkill(OVOSSkill):
     """Home Assistant skill for Neon OS. Requires the PHAL Home Assistant plugin."""
 
+    _connected = False
+    connected_intents = (
+            "sensor.intent",
+            "turn.on.intent",
+            "turn.off.intent",
+            "stop.intent",
+            "lights.get.brightness.intent",
+            "lights.set.brightness.intent",
+            "lights.increase.brightness.intent",
+            "lights.decrease.brightness.intent",
+            "lights.get.color.intent",
+            "lights.set.color.intent",
+            "show.area.dashboard.intent",
+            "assist.intent",
+        )
+
+    @property
+    def connected(self):
+        if self._connected is False:
+            self._connected = self.on_ready(Message(
+                        "ovos.phal.plugin.homeassistant.check_connected",
+                        None,
+                        {"skill_id": self.skill_id}
+                    )
+                )
+        return self._connected
+
     def initialize(self):
+        try:
+            assert self.connected is True
+        except AssertionError:
+            self.log.warning("Home Assistant PHAL plugin not installed or not running!")
+        if not self.connected:
+            self.on_ready(Message("ovos.phal.plugin.homeassistant.check_connected", None, {"skill_id": self.skill_id}))
+        if self.connected:
+            self.enable_ha_intents()
```

Comment:
Consider running this in a handler for `mycroft.ready` so there's a better chance of PHAL plugin being started

Reply:
I removed the handler for `mycroft.ready` because I wasn't sure it was guaranteed to only return once. How often does `mycroft.ready` emit?